src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -185,18 +189,41 @@ public static void onPlayerDestroyItem(Player player, @NotNull ItemStack stack,
         MinecraftForge.EVENT_BUS.post(new PlayerDestroyItemEvent(player, stack, hand));
     }
 
-    public static Result canEntitySpawn(Mob entity, LevelAccessor level, double x, double y, double z, BaseSpawner spawner, MobSpawnType spawnReason)
+    /**
+     * Vanilla calls to {@link Mob#finalizeSpawn} are replaced with calls to this method via coremod.<p>
+     * Mods should call this method in place of calling {@link Mob#finalizeSpawn}. Super calls (from within overrides) should not be wrapped.
```

Comment:
So to clarify, modders should still override finalizeSpawn for their own mobs as the coremod will automatically wrap vanilla calls of it, and if they ever manually call finalizeSpawn anywhere in their implementation, they should change it to call this hook instead?

Reply:
Correct.